packages/components/src/internal/components/editable/utils.test.ts

```diff
@@ -637,6 +562,216 @@ describe('getUpdatedDataFromGrid', () => {
+            value: '2020-12-23',
+        });
+        expect(getValidatedEditableGridValue(new Date('2020-12-23 14:34'), dateCol)).toStrictEqual({
+            message: undefined,
+            value: '2020-12-23',
+        });
+    });
+
+    test('invalid date value', () => {
+        expect(getValidatedEditableGridValue('BOGUS', dateCol)).toStrictEqual({
+            message: 'Invalid date',
+            value: 'BOGUS',
+        });
+        expect(getValidatedEditableGridValue(true, dateCol)).toStrictEqual({ message: 'Invalid date', value: true });
+        expect(getValidatedEditableGridValue(13, dateCol).message).toBe(undefined);
+        expect(getValidatedEditableGridValue('2020-12-INVALID 14:34', dateCol)).toStrictEqual({
+            message: 'Invalid date',
+            value: '2020-12-INVALID 14:34',
+        });
+        expect(getValidatedEditableGridValue('2020-13-23 14:34', dateCol)).toStrictEqual({
+            message: 'Invalid date',
+            value: '2020-13-23 14:34',
+        });
+        expect(getValidatedEditableGridValue(new Date('2020-13-23 14:34'), dateCol).message).toBe('Invalid date');
+    });
+
+    test('valid dateTimeCol value', () => {
+        expect(getValidatedEditableGridValue('2020-12-23', dateTimeCol)).toStrictEqual({
+            message: undefined,
+            value: '2020-12-23 00:00:00',
+        });
+        expect(getValidatedEditableGridValue('2020-12-23 00:00:00', dateTimeCol)).toStrictEqual({
+            message: undefined,
+            value: '2020-12-23 00:00:00',
+        });
+        expect(getValidatedEditableGridValue('2020-12-23 14:34', dateTimeCol)).toStrictEqual({
+            message: undefined,
+            value: '2020-12-23 14:34:00',
+        });
+        expect(getValidatedEditableGridValue(new Date('2020-12-23 14:34'), dateTimeCol)).toStrictEqual({
+            message: undefined,
+            value: '2020-12-23 14:34:00',
+        });
+    });
+
+    test('invalid dateTimeCol value', () => {
+        expect(getValidatedEditableGridValue('BOGUS', dateTimeCol)).toStrictEqual({
+            message: 'Invalid date time',
+            value: 'BOGUS',
+        });
+        expect(getValidatedEditableGridValue(true, dateTimeCol)).toStrictEqual({
+            message: 'Invalid date time',
+            value: true,
+        });
+        expect(getValidatedEditableGridValue(13, dateTimeCol).message).toBe(undefined);
+        expect(getValidatedEditableGridValue('2020-12-INVALID 14:34', dateTimeCol)).toStrictEqual({
+            message: 'Invalid date time',
+            value: '2020-12-INVALID 14:34',
+        });
+        expect(getValidatedEditableGridValue('2020-13-23 14:34', dateTimeCol)).toStrictEqual({
+            message: 'Invalid date time',
+            value: '2020-13-23 14:34',
+        });
+        expect(getValidatedEditableGridValue(new Date('2020-13-23 14:34'), dateTimeCol).message).toBe(
+            'Invalid date time'
+        );
+    });
```

Comment:
Is it worth having a test specific to time fields as well?

Reply:
We don't yet do any client side validation / conversions for time fields (this will be coming with the next story for grid validation). But I'll add an array of col types to iterate over for this 'non-date columns' test case.

Reply:
Following up here. Time field test coverage was added with #1527.